flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CatalogBaseTable.java

```diff
@@ -18,18 +18,32 @@
 
 package org.apache.flink.table.catalog;
 
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.table.api.Schema;
 import org.apache.flink.table.api.TableSchema;
 import org.apache.flink.table.factories.DynamicTableFactory;
 
 import java.util.Map;
 import java.util.Optional;
 
 /**
- * CatalogBaseTable is the common parent of table and view. It has a map of key-value pairs defining
- * the properties of the table.
+ * A common parent that describes the <i>unresolved</i> metadata of a table or view in a catalog.
+ *
+ * @see CatalogTable
+ * @see CatalogView
  */
+@PublicEvolving
 public interface CatalogBaseTable {
 
+    /** The kind of {@link CatalogBaseTable}. */
+    enum TableKind {
```

Comment:
Is "TableKind" the correct abstraction to use if one value is "Table"? If we e.g. enumerate types of apples, then AppleKind.GRANNY_SMITH is reasonable, but AppleKind.APPLE sounds out of place. A "table" is not a "kind of table".

I sort of feel the same way here. `CatalogBaseTable` has that problem, too, but that class existed already, so we probably can't rename it easily now.

Reply:
We were also discussing about the name in the past but couldn't come up with a better name.

Some background: a view is commonly defined as a `virtual table`. See https://www.geeksforgeeks.org/sql-views/ or https://www.w3schools.com/sql/sql_ref_create_view.asp or Wikipedia. So we have to distinguish between a `real/base table` (= table) or `virtual table` (=view). The common name is still table unless you have a better idea :(